content/en/docs/concepts/storage/persistent-volumes.md

```diff
@@ -308,28 +308,31 @@ If expanding underlying storage fails, the cluster administrator can manually re
 
 ## Types of Persistent Volumes
 
-PersistentVolume types are implemented as plugins.  Kubernetes currently supports the following plugins:
-
-* GCEPersistentDisk
-* AWSElasticBlockStore
-* AzureFile
-* AzureDisk
-* CSI
-* FC (Fibre Channel)
-* FlexVolume
-* Flocker
-* NFS
-* iSCSI
-* RBD (Ceph Block Device)
-* CephFS
-* Cinder (OpenStack block storage)
-* Glusterfs
-* VsphereVolume
-* Quobyte Volumes
-* HostPath (Single node testing only -- local storage is not supported in any way and WILL NOT WORK in a multi-node cluster)
-* Portworx Volumes
-* ScaleIO Volumes
-* StorageOS
+PersistentVolume types are implemented as plugins. Kubernetes currently supports the following plugins:
+
+* [`awsElasticBlockStore`](/docs/concepts/storage/volumes/#awselasticblockstore) - AWS Elastic Block Store (EBS)
+* [`azureDisk`](/docs/concepts/sotrage/volumes/#azuredisk) - Azure Disk
+* [`azureFile`](/docs/concepts/storage/volumes/#azurefile) - Azure File
+* [`cephfs`](/docs/concepts/storage/volumes/#cephfs) - CephFS volume
+* [`cinder`](/docs/concepts/storage/volumes/#cinder) - Cinder (OpenStack block storage)
+* [`csi`](/docs/concepts/storage/volumes/#csi) - Container Storage Interface (CSI)
+* [`fc`](/docs/concepts/storage/volumes/#fc) - Fibre Channel (FC) storage
+* [`flexVolume`](/docs/concepts/storage/volumes/#flexVolume) - FlexVolume
+* [`flocker`](/docs/concepts/storage/volumes/#flocker) - Flocker storage
+* [`gcePersistentDisk`](/docs/concepts/storage/volumes/#gcepersistentdisk) - GCE Persistent Disk
+* [`glusterfs`](/docs/concepts/storage/volumes/#glusterfs) - Glusterfs volume
+* [`hostPath`](/docs/concepts/storage/volumes/#hostpath) - HostPath volume
+  (for single node testing only; local storage is not supported in any way **does not work**
+  in a multi-node cluster)
+* [`iscsi`](/docs/concepts/storage/volumes/#iscsi) - iSCSI (SCSI over IP) storage
+* [`nfs`](/docs/concepts/storage/volumes/#nfs) - Network File System (NFS) storage
+* `photonPersistentDisk` - Photon controller persistent disk
+* [`portworxVolume`](/docs/concepts/storage/volumes/#portworxvolume) - Portworx volume
+* [`quobyte`](/docs/concepts/storage/volumes/#quobyte) - Quobyte volume
+* [`rbd`](/docs/concepts/storage/volumes/#rbd) - Rados Block Device (RBD) volume
+* [`scaleIO`](/docs/concepts/storage/volumes/#scaleio) - ScaleIO volume
```

Comment:
Should we add warnings for deprecated plugins? https://github.com/kubernetes/kubernetes/pull/80099

Also a number of cloud plugins have been deprecated and replaced with csi: aws ebs, gce pd, azure disk, vsphere

Reply:
According to [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L72-L75), we have only deprecated `cinder` and `scaleio` as volume plugins. Please let me know if this is captured elsewhere in the source.

Reply:
As for `photonPersistentDisk`, I believe it is still a valid volume type to use though the cloud provider has been deprecaated.

Reply:
For photon, the cloud provider has been removed, so it will not function at all. The API was left because we cannot remove fields from the v1 API.

The cloud plugins are deprecated as part of the csi migration features. There have been release notes about them in previous releases but is missing from the documentation. For example:
https://github.com/kubernetes/kubernetes/pull/90896
https://github.com/kubernetes/kubernetes/pull/85637
https://github.com/kubernetes/kubernetes/pull/85231
https://github.com/kubernetes/kubernetes/pull/85237

Also, on this page, flocker has been marked as deprecated: https://kubernetes.io/docs/concepts/storage/volumes/#flocker

Reply:
@msau42 I'm okay with the removal `photonPersistentDisk` if it will not function at all. However, I'm a little worried that we are dropping a feature while leaving the API pieces out there. What a user will get if he/she is using this volume type?

As for the `azureDisk` (or maybe `azureFile`), `cinder`, `gcePersistentDisk`, `awsElasticBlockStore`, we are talking about different aspects of the migration to CSI, I guess. Please help confirm that the migration to CSI automatically deprecates the in-tree plugins. From your description, I guess the answer is yes. I am not so sure because most of the time, we have an option B does not automatically mean that option A is deprecated. If we ARE officially marking them as deprecated, we may want to spread this information all over the places, i.e. the source file, the API spec and the docs.

From docs' perspective, I want to follow up this in another PR. Sounds good? We will need to revise the docs/concepts/storage/volumes.md file anyway.

Reply:
I am fine with documenting deprecated plugins in a separate PR. Thanks!

/lgtm